tests/Functional/Schema/MySQL/ComparatorTest.php

```diff
@@ -136,6 +136,30 @@ public function testChangeColumnCollation(): void
         ComparatorTestUtils::assertDiffNotEmpty($this->connection, $this->comparator, $table);
     }
 
+    public function testImplicitColumnCharset(): void
+    {
+        self::markTestIncomplete('See https://github.com/doctrine/dbal/issues/5338');
```

Comment:
The test passes on `3.x` but fails on `4.0.x` due to the removal of `utf8` as the default table charset in https://github.com/doctrine/dbal/pull/4644.

If `utf8` below is replaced with `utf8mb4` in `3.x` the test will fail there as well:

https://github.com/doctrine/dbal/blob/a18ffc1a1fde506b17c34e9523fce95f16838359/src/Platforms/AbstractMySQLPlatform.php#L449-L451

I've reopened the issue and will try to improve the fix.